docs/develop/typescript/data-encryption.mdx

```diff
@@ -11,41 +11,129 @@ tags:
 description: Learn how to create a custom Payload Converter in TypeScript with Temporal SDKs to handle non-JSON-serializable values, configure your Data Converter, and utilize protobufs and encryption seamlessly in your Workflows and Activities.
 ---
 
-## How to use a custom payload converter in TypeScript {#custom-payload-conversion}
+## Payload Converter and Payload Codec Summary
 
-Temporal SDKs provide a [Payload Converter](/dataconversion#payload-converter) that can be customized to convert a custom data type to a [Payload](/dataconversion#payload) and back.
+This section will summarize the difference between a Payload Converter and Payload Codec.
+
+### Payload Converter
 
-Implementing custom Payload conversion is optional.
-It is needed only if the [default Data Converter](/dataconversion#default-data-converter) does not support your custom values.
+Payload Converters are responsible for serializing application objects into a Payload and deserializing them back into application objects. A Payload, in this context, is a binary form suitable for network transmission that may include some metadata. This serialization process transforms an object (like those in JSON or Protobuf formats) into a binary format and vice versa. For example, an object might be serialized to JSON with UTF-8 byte encoding or to a protobuf binary using a specific set of protobuf message definitions.
```

Comment:
I don't think you need to use "are responsible for". I think they "serialize application objects" (assuming objects is the correct phrase for data types here). You capitalized Payload, which is not a reserved Temporal term. I think the plural "into payloads, and deserialize them back" also makes the writing smoother. 

I like the definition of the payload. Consider splitting metadata into a separate thought. The next sentence "This serialization process" feels redundant to me. You might put the focus on JSON or Protobuf instead.  JSON, with its human-readable structure, is a common data interchange solution for web APIs due to its simplicity and ease of integration. Protobufs offer efficient binary serialized formats that reduce payload sizes and support fast processing. That's ideal for high-performance, etc etc. 

What I don't see is how this connects to the next line though. Serialization/deserialization is non-deterministic, so converters can live in Workflows, but encryption and other types of novel encoding, cannot, right? So there needs to be a second piece that can handle non-deterministic logic, which lives outside the Workflow, if I'm understanding this correctly. And it would help to have words here that explain the difference.

"Because of this" is I think better than "Consequently", we prefer "can't" to "cannot", "use" to "employ". And then I think a breath is called for. Maybe new sentence "This excludes (most) encryption because encryption is almost always non-determ."

And then there's the missing piece that leads to "Well, what do you do with the part that can't live inside the Workflow", a transition that gets you from converters to codecs.